Source/Core/InputCommon/ControllerInterface/CemuHookUDPServer/CemuHookUDPServer.cpp

```diff
@@ -0,0 +1,355 @@
+
+  Config::SetBaseOrCurrent(Settings::SERVER_ENABLED, s_server_enabled);
+  Config::SetBaseOrCurrent(Settings::SERVER_ADDRESS, s_server_address);
+  Config::SetBaseOrCurrent(Settings::SERVER_PORT, s_server_port);
+
+  Config::Save();
+}
+
+Device::Device(Proto::DsModel model, int index)
+    : m_model(model), m_index(index),
+      m_client_uid(Common::Random::GenerateValue<u32>()), m_accel{}, m_gyro{},
+      m_next_reregister(std::chrono::steady_clock::time_point::min()), m_pad_data{}, m_prev_touch{},
+      m_prev_touch_valid(false), m_touch_x(0), m_touch_y(0)
+{
+  m_socket.setBlocking(false);
+
+  AddInput(new AnalogInput("Pad W", m_pad_data.button_dpad_left_analog, 255));
+  AddInput(new AnalogInput("Pad S", m_pad_data.button_dpad_down_analog, 255));
+  AddInput(new AnalogInput("Pad E", m_pad_data.button_dpad_right_analog, 255));
+  AddInput(new AnalogInput("Pad N", m_pad_data.button_dpad_up_analog, 255));
+  AddInput(new AnalogInput("Square", m_pad_data.button_square_analog, 255));
+  AddInput(new AnalogInput("Cross", m_pad_data.button_cross_analog, 255));
+  AddInput(new AnalogInput("Circle", m_pad_data.button_circle_analog, 255));
+  AddInput(new AnalogInput("Triangle", m_pad_data.button_triangle_analog, 255));
+  AddInput(new AnalogInput("L1", m_pad_data.button_l1_analog, 255));
+  AddInput(new AnalogInput("R1", m_pad_data.button_r1_analog, 255));
+
+  AddInput(new AnalogInput("L2", m_pad_data.trigger_l2, 255));
+  AddInput(new AnalogInput("R2", m_pad_data.trigger_r2, 255));
+
+  AddInput(new Button("L3", m_pad_data.button_states1, 0x2));
+  AddInput(new Button("R3", m_pad_data.button_states1, 0x4));
+  AddInput(new Button("Share", m_pad_data.button_states1, 0x1));
+  AddInput(new Button("Options", m_pad_data.button_states1, 0x8));
+  AddInput(new Button("PS", m_pad_data.button_ps, 0x1));
+  AddInput(new Button("Touch Button", m_pad_data.button_touch, 0x1));
+
+  AddInput(new AnalogInput("Left X-", m_pad_data.left_stick_x, -128, -128));
+  AddInput(new AnalogInput("Left X+", m_pad_data.left_stick_x, 127, -128));
+  AddInput(new AnalogInput("Left Y-", m_pad_data.left_stick_y_inverted, -128, -128));
+  AddInput(new AnalogInput("Left Y+", m_pad_data.left_stick_y_inverted, 127, -128));
+  AddInput(new AnalogInput("Right X-", m_pad_data.right_stick_x, -128, -128));
+  AddInput(new AnalogInput("Right X+", m_pad_data.right_stick_x, 127, -128));
+  AddInput(new AnalogInput("Right Y-", m_pad_data.right_stick_y_inverted, -128, -128));
+  AddInput(new AnalogInput("Right Y+", m_pad_data.right_stick_y_inverted, 127, -128));
+
+  AddInput(new AnalogInput("Touch X-", m_touch_x, -TOUCH_X_AXIS_MAX));
+  AddInput(new AnalogInput("Touch X+", m_touch_x, TOUCH_X_AXIS_MAX));
+  AddInput(new AnalogInput("Touch Y-", m_touch_y, -TOUCH_Y_AXIS_MAX));
+  AddInput(new AnalogInput("Touch Y+", m_touch_y, TOUCH_Y_AXIS_MAX));
+
+  AddInput(new AnalogInput("Accelerometer Left", m_accel.x, -WiimoteEmu::ACCELEROMETER_MAX));
```

Comment:
ControllerInterface can handle values larger than 1.0. (Specifically for when this situation of unbounded acceleration/rotational-velocity inputs came up) I suggest you pass through the raw values instead of dividing by an arbitrary "MAX" and then later multiplying by it. You should be able to eliminate those magic numbers entirely.

Reply:
That's really cool. I'll do that.